sqlmesh/core/context.py

```diff
@@ -1335,6 +1335,56 @@ def plan_builder(
         Returns:
             The plan builder.
         """
+        kwargs: t.Dict[
+            str,
+            t.Union[
+                t.Optional[TimeLike],
+                t.Optional[str],
+                t.Optional[bool],
+                t.Optional[t.Iterable[str]],
+                t.Optional[t.Collection[str]],
+            ],
+        ] = {
+            "start": start,
+            "end": end,
+            "execution_time": execution_time,
+            "create_from": create_from,
+            "skip_tests": skip_tests,
+            "restate_models": restate_models,
+            "no_gaps": no_gaps,
+            "skip_backfill": skip_backfill,
+            "empty_backfill": empty_backfill,
+            "forward_only": forward_only,
+            "allow_destructive_models": allow_destructive_models,
+            "no_auto_categorization": no_auto_categorization,
+            "effective_from": effective_from,
+            "include_unmodified": include_unmodified,
+            "select_models": select_models,
+            "backfill_models": backfill_models,
+            "enable_preview": enable_preview,
+            "run": run,
+            "diff_rendered": diff_rendered,
+            "skip_linter": skip_linter,
+        }
+        user_defined_flags: t.Dict[
+            str,
+            t.Union[
+                t.Optional[TimeLike],
+                t.Optional[str],
+                t.Optional[bool],
+                t.Optional[t.Iterable[str]],
+                t.Optional[t.Collection[str]],
+            ],
+        ] = {k: v for k, v in kwargs.items() if v is not None}
```

Comment:
When we discussed, I thought we were going to put this logic in the plan builder class, not in the method. Can you please clarify with @izeigerman which he was referring too.

Also, I'm not so sure the union is useful over any. Especially if it has to be defined in multiple places.